File: ios/SwiftUIProject/ReactNative/Modules/Router/ReactNativeRouterModule.m
//
//  ReactNativeRouterModule.m
//  SwiftUIProject
//
//  Created by ralph-L on 2023/09/01.
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(ReactNativeRouterModule, NSObject)

RCT_EXTERN_METHOD(push:(NSString *)path)
RCT_EXTERN_METHOD(goBack)
RCT_EXTERN_METHOD(popToRoot)

@end

File: ios/SwiftUIProject/ReactNative/Modules/Router/ReactNativeRouterModule.swift
//
//  ReactNativeRouterModule.swift
//  SwiftUIProject
//
//  Created by ralph-L on 2023/09/01.
//

import Foundation

@objc(ReactNativeRouterModule)
final class ReactNativeRouterModule: NSObject {
    @objc(push:)
    func push(_ path: String) {
        DispatchQueue.main.async {
            Router.shared.push(path: path)
        }
    }

    @objc
    func goBack() {
        DispatchQueue.main.async {
            Router.shared.goBack()
        }
    }
    
    @objc
    func popToRoot() {
        DispatchQueue.main.async {
            Router.shared.popToRoot()
        }
    }

    @objc
    static func requiresMainQueueSetup() -> Bool {
        false
    }
}
